spec/controllers/alchemy/api/pages_controller_spec.rb

```diff
@@ -48,6 +48,107 @@ module Alchemy
       end
     end
 
+    describe '#nested' do
+      let!(:page) { create(:alchemy_page, :public, page_layout: 'contact') }
+
+      it "returns all pages as nested json tree without admin related infos", :aggregate_failures do
+        alchemy_get :nested, format: :json
+
+        expect(response.status).to eq(200)
+        expect(response.content_type).to eq('application/json')
+
+        result = JSON.parse(response.body)
+
+        expect(result).to have_key('pages')
+        expect(result['pages'].size).to eq(1)
+        expect(result['pages'][0]).to have_key('children')
+        expect(result['pages'][0]['children'].size).to eq(1)
+
+        child = result['pages'][0]['children'][0]
+
+        expect(child['name']).to eq(page.name)
+        expect(child).to_not have_key('definition_missing')
+        expect(child).to_not have_key('folded')
+        expect(child).to_not have_key('locked')
+        expect(child).to_not have_key('permissions')
+        expect(child).to_not have_key('status_titles')
+      end
+
+      context "as author" do
+        before do
+          authorize_user(build(:alchemy_dummy_user, :as_author))
+        end
+
+        it "returns all pages as nested json tree with admin related infos", :aggregate_failures do
+          alchemy_get :nested, format: :json
+
+          expect(response.status).to eq(200)
+          expect(response.content_type).to eq('application/json')
+
+          result = JSON.parse(response.body)
+
+          expect(result).to have_key('pages')
+          expect(result['pages'].size).to eq(1)
+          expect(result['pages'][0]).to have_key('children')
+          expect(result['pages'][0]['children'].size).to eq(1)
+
+          child = result['pages'][0]['children'][0]
+
+          expect(child['name']).to eq(page.name)
+          expect(child).to have_key('definition_missing')
+          expect(child).to have_key('folded')
+          expect(child).to have_key('locked')
+          expect(child).to have_key('permissions')
+          expect(child).to have_key('status_titles')
+        end
+      end
+
+      context "when a page_id is passed" do
+        it 'returns all pages as nested json from this page only' do
+          alchemy_get :nested, page_id: page.id, format: :json
+
+          expect(response.status).to eq(200)
+          expect(response.content_type).to eq('application/json')
+
+          result = JSON.parse(response.body)
+
+          expect(result).to have_key('pages')
+          expect(result['pages'][0]['name']).to eq(page.name)
+        end
+      end
+
+      context "when `elements=true` is passed" do
+        it 'returns all pages as nested json tree with elements included' do
+          alchemy_get :nested, elements: 'true', format: :json
+
+          expect(response.status).to eq(200)
+          expect(response.content_type).to eq('application/json')
+
+          result = JSON.parse(response.body)
+
+          expect(result).to have_key('pages')
```

Comment:
There is the possibility to share some of the expectations using shared examples. But I don't see the need to do that within this PR right now.